pkg/controller/status/status.go

```diff
@@ -252,71 +289,15 @@ func (c *Controller) merge(existing *configv1.ClusterOperator) *configv1.Cluster
 		// once we've initialized set Failing and Disabled as best we know
-		if len(disabledMessage) > 0 {
-			setOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
-				Type:    OperatorDisabled,
-				Status:  configv1.ConditionTrue,
-				Reason:  disabledReason,
-				Message: disabledMessage,
-			})
-		} else {
-			setOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
-				Type:   OperatorDisabled,
-				Status: configv1.ConditionFalse,
-				Reason: "AsExpected",
-			})
-		}
-
-		if len(errorMessage) > 0 {
-			klog.V(4).Infof("The operator has some internal errors: %s", errorMessage)
-			setOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
-				Type:               configv1.OperatorDegraded,
-				Status:             configv1.ConditionTrue,
-				LastTransitionTime: metav1.Time{Time: last},
-				Reason:             reason,
-				Message:            errorMessage,
-			})
-		} else {
-			setOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
-				Type:   configv1.OperatorDegraded,
-				Status: configv1.ConditionFalse,
-				Reason: "AsExpected",
-			})
-		}
-
-		if len(uploadErrorReason) > 0 {
-			setOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
-				Type:               UploadDegraded,
-				Status:             configv1.ConditionTrue,
-				LastTransitionTime: metav1.Time{Time: last},
-				Reason:             uploadErrorReason,
-				Message:            uploadErrorMessage,
-			})
-		} else {
-			removeOperatorStatusCondition(&existing.Status.Conditions, UploadDegraded)
-		}
-
-		if len(downloadReason) > 0 {
-			setOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
-				Type:               InsightsDownloadDegraded,
-				Status:             configv1.ConditionTrue,
-				LastTransitionTime: metav1.Time{Time: last},
-				Reason:             downloadReason,
-				Message:            downloadMessage,
-			})
-		} else {
-			removeOperatorStatusCondition(&existing.Status.Conditions, InsightsDownloadDegraded)
-		}
+		setDisabledOperatorStatusCondition(existing, disabledReason, disabledMessage)
+		setDegradedOperatorStatusCondition(existing, last, errorReason, errorMessage)
+		setUploadDegradedOperatorStatusCondition(existing, last, uploadErrorReason, uploadErrorMessage)
+		setInsightsDownloadDegradedOperatorStatusCondition(existing, last, downloadReason, downloadMessage)
 	}
+}
 
-	// once the operator is running it is always considered available
-	setOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
-		Type:   configv1.OperatorAvailable,
-		Status: configv1.ConditionTrue,
-		Reason: "AsExpected",
-	})
-
-	// update the Progressing condition with a summary of the current state
+func (c *Controller) updateProcessingConditionWithSummary(existing *configv1.ClusterOperator,
```

Comment:
It looks like it doesn't have to be a receiver method. Same for the `updateDisabledAndFailingConditions`

Reply:
Agreed. In the beginning, I was planning to move the `last` to the struct properties, but... after doing it... well, it doesn't seem right.